server/src/main/java/com/linecorp/centraldogma/server/internal/api/converter/WatchRequestConverter.java

```diff
@@ -81,6 +81,21 @@ public WatchRequest convertRequest(
         return new WatchRequest(lastKnownRevision, timeoutMillis, notifyEntryNotFound);
     }
 
+    String extractRevision(String ifNoneMatch) {
+        // Three below cases are valid:
+        // - <revision> (for backward compatibility)
+        // - "<revision>"
+        // - W/"<revision>"
+        if (ifNoneMatch.startsWith("\"") && ifNoneMatch.endsWith("\"") && !ifNoneMatch.equals("\"")) {
```

Comment:
How about?
```java
if (ifNoneMatch.length() > 1 && ifNoneMatch.charAt(0) == '"' &&
    ifNoneMatch.charAt(ifNoneMatch.length() - 1) == '"') {
    return ...
}

if (ifNoneMatch.length() > 2 && ifNoneMatch.startsWith("W/\"") &&
    ifNoneMatch.charAt(ifNoneMatch.length() - 1) == '"') {
    return ...
}
```

Reply:
More simple 🙂

Reply:
And to handle, `""` I revise condition `ifNoneMatch.length() > 1` to `ifNoneMatch.length() > 2`.